test/Microsoft.NET.Publish.Tests/GivenThatWeWantToPublishAHelloWorldProject.cs

```diff
@@ -162,6 +164,123 @@ public static void Main()
         }
 
         [Fact]
+        public void Conflicts_are_resolved_when_publishing_a_portable_app()
+        {
+            Conflicts_are_resolved_when_publishing(false);
+        }
+
+        [Fact]
+        public void Conflicts_are_resolved_when_publishing_a_self_contained_app()
+        {
+            Conflicts_are_resolved_when_publishing(true);
+        }
+
+        void Conflicts_are_resolved_when_publishing(bool selfContained, [CallerMemberName] string callingMethod = "")
+        {
+            var targetFramework = "netcoreapp2.0";
+            var rid = selfContained ? EnvironmentInfo.GetCompatibleRid(targetFramework) : null;
```

Comment:
It would be good to also test "low fat apps" as well.  See https://github.com/dotnet/sdk/pull/1053.